src/Common/src/Common.Security/LocalCertificateWriter.cs

```diff
@@ -20,118 +20,122 @@ public sealed class LocalCertificateWriter
 
-        // Create Root CA and intermediate cert PFX with private key (if not already there)
+        // Create a directory a level above the running project to contain the root and intermediate certificates
         if (!Directory.Exists(Path.Combine(ParentPath, "GeneratedCertificates")))
         {
             Directory.CreateDirectory(Path.Combine(ParentPath, "GeneratedCertificates"));
         }
 
+        // Create the root certificate if it doesn't already exist (can be shared by multiple applications)
         if (!File.Exists(RootCaPfxPath))
         {
-            caCertificate = CreateRoot("CN=SteeltoeGeneratedCA");
+            caCertificate = CreateRootCertificate("CN=SteeltoeGeneratedCA");
             File.WriteAllBytes(RootCaPfxPath, caCertificate.Export(X509ContentType.Pfx));
         }
         else
         {
             caCertificate = new X509Certificate2(RootCaPfxPath);
         }
 
+        // Create the intermediate certificate if it doesn't already exist (can be shared by multiple applications)
         X509Certificate2 intermediateCertificate;
 
-        // Create intermediate cert PFX with private key (if not already there)
         if (!File.Exists(IntermediatePfxPath))
         {
-            intermediateCertificate = CreateIntermediate("CN=SteeltoeGeneratedIntermediate", caCertificate);
+            intermediateCertificate = CreateIntermediateCertificate("CN=SteeltoeGeneratedIntermediate", caCertificate);
             File.WriteAllBytes(IntermediatePfxPath, intermediateCertificate.Export(X509ContentType.Pfx));
         }
         else
         {
             intermediateCertificate = new X509Certificate2(IntermediatePfxPath);
         }
 
-        X509Certificate2 clientCertificate = CreateClient(subject, intermediateCertificate, new SubjectAlternativeNameBuilder());
+        var subjectAlternativeNameBuilder = new SubjectAlternativeNameBuilder();
+
+        X509Certificate2 clientCertificate = CreateClientCertificate(subject, intermediateCertificate, subjectAlternativeNameBuilder);
 
         // Create a folder inside the project to store generated certificate files
         if (!Directory.Exists(Path.Combine(AppBasePath, "GeneratedCertificates")))
         {
             Directory.CreateDirectory(Path.Combine(AppBasePath, "GeneratedCertificates"));
         }
 
-        string certContents = "-----BEGIN CERTIFICATE-----\r\n" +
-            Convert.ToBase64String(clientCertificate.Export(X509ContentType.Cert), Base64FormattingOptions.InsertLineBreaks) +
-            "\r\n-----END CERTIFICATE-----\r\n" + "-----BEGIN CERTIFICATE-----\r\n" +
-            Convert.ToBase64String(intermediateCertificate.Export(X509ContentType.Cert), Base64FormattingOptions.InsertLineBreaks) +
-            "\r\n-----END CERTIFICATE-----\r\n";
+#if NET8_0_OR_GREATER
+        string chainedCertificateContents = clientCertificate.ExportCertificatePem() + "\r\n" + intermediateCertificate.ExportCertificatePem();
```

Comment:
To be honest, I'm not sure. I can't see why it would matter though as these should only ever be written and then read from the same OS. This will affect the line formatting quite a bit too